app/Helpers/Payment/EventPayment.php

```diff
@@ -0,0 +1,62 @@
+<?php
+namespace App\Helpers\Payment;
+
+use App\Event;
+use App\Participant;
+
+class EventPayment implements PaymentInterface
+{
+    private $event;
+    private $participant;
+    private $existing;
+
+    public function event(Event $event)
+    {
+        $this->event = $event;
+        return $this;
+    }
+
+    public function participant(Participant $part)
+    {
+        $this->participant = $part;
+        return $this;
+    }
+
+    public function existing(bool $existing)
+    {
+        $this->existing = $existing;
+        return $this;
+    }
+
+    public function getTotalAmount(): float
+    {
+        return round(($this->participant->incomeBasedDiscount() * $this->event->prijs) / 5) * 5;
+    }
+
+    public function getDescription(): string
+    {
+        return $this->event->code . " - " . str_replace("  ", " ", $this->participant->voornaam . " " . $this->participant->tussenvoegsel . " " . $this->participant->achternaam);
```

Comment:
Je kunt het hele stuk van de naam van de deelnemer `str_replace(...)` vervangen door `$this->participant->volnaam`. Is een custom getter op het `Participant` model.

Reply:
Check die was voor mij nog onbekend, willdo